src/coreclr/pal/src/exception/signal.cpp

```diff
@@ -690,6 +723,11 @@ static void sigterm_handler(int code, siginfo_t *siginfo, void *context)
         _ASSERTE(g_pSynchronizationManager != nullptr);
 
         g_pSynchronizationManager->SendTerminationRequestToWorkerThread();
+
+        if (g_enableDumpOnSigTerm)
```

Comment:
What is the scenario when we want a dump on SIGTERM? It seems problematic, as the dump generation will now run in parallel with the regular SIGTERM handling. The SendTerminationRequestToWorkerThread just pings another thread and returns.

Reply:
The point of these changes to get parity with Mono for VSMac when they start running on .NET Core.  And Mono (I'm double checking - see the issue above) generates their crash info json file on SIGTERM. 

You did point out a race that I'll look into.  I'll probably have to move this new code to TerminationRequestHandlingRoutine() in syncmanager.cpp which runs on that termination thread.

Reply:
SIGTERM is currently handled by the runtime by ultimately invoking HandleTerminationRequest in the runtime. That calls ForceEEShutdown(SCA_ExitProcessWhenShutdownComplete); which invokes the Process.Exited event and then shuts down the process. So to avoid disrupting that process, it seems the dump should not be generated before the Process.Exited returns.

Reply:
> The point of these changes to get parity with Mono for VSMac when they start running on .NET Core.

Is configuring this via env variable going to work for VSMac? I see two potential problems: 
- How to set it before the VSMac process starts
- How to clear it for all child processes

Reply:
I was definitely going to move it after all the of is called but ForceEEShutdown doesn't return because it ends up calling ExitProcess.  I need to think about the best way to handle this.

Reply:
They already have to set the other dump generation (i.e. COMPlus_DbgEnableMiniDump, etc.+)  env vars so I didn't think one more would be a big deal.

Reply:
@kdubau how important is generating a dump on SIGTERM? It is more difficult to implement that I thought.

Reply:
I think what I just pushed allows SIGTERM to generate coredumps (when  COMPlus_EnableDumpOnSigTerm=1) without the problems pointed out above.